CRM/Core/BAO/UFField.php

```diff
@@ -119,15 +168,21 @@ public static function del($id) {
    *
    * @param array $params
    *   An associative array with field and values.
-   * @param $ids
-   *
-   * @return mixed
-   * @ids   array $ids    array that containd ids
    *
+   * @return bool
    */
-  public static function duplicateField($params, $ids) {
+  public static function duplicateField($params) {
     $ufField = new CRM_Core_DAO_UFField();
-    $ufField->uf_group_id = CRM_Utils_Array::value('uf_group', $ids);
+
+    $ufField->uf_group_id = CRM_Utils_Array::value('uf_group_id', $params);
+
+    $field_type       = CRM_Utils_Array::value('field_type', $params);
+    $field_name       = CRM_Utils_Array::value('field_name', $params);
+    $location_type_id = CRM_Utils_Array::value('location_type_id', $params, CRM_Utils_Array::value('website_type_id', $params));
+    $phone_type       = CRM_Utils_Array::value('phone_type_id', $params, CRM_Utils_Array::value('phone_type', $params));
+
+    $params['field_name'] = array($field_type, $field_name, $location_type_id, $phone_type);
```

Comment:
Declaring 'field_name' here as an array feels kind of hack-ish and we can probably do without it. However since the parameter passed to this function was not a reference to the $params array, this should not cause any problems.